src/DeviceInterface/InterfacePrototype.js

```diff
@@ -394,7 +394,7 @@ class InterfacePrototype {
                 if (Array.isArray(data.credentials) && data.credentials.length > 0) {
                     return data.credentials
                 } else {
-                    return this.getLocalCredentials()
+                    return this.getLocalCredentials().filter(cred => !!cred[subtype] || subtype === 'password' || cred.id === PROVIDER_LOCKED)
```

Comment:
This ensures certain items are filtered out. For example, if we are in a username field but there's an entry without a username, we don't show it.